### PR TITLE
React-native-windows-init reuses existing projectGUIDs by default

### LINKED ISSUE #8621: react-native-windows-init should re-use existing project guids if present by default
Right now the CLI generates a new project guid, which updates not only means an unnecessary change in the project file, but also lots of changes in the solution file.

This may be a non-issue for app projects, but now that we have lib projects, this causes problems.

Namely, the change in project guid in a native module will cause all app consumers to have update their project and solution files. Autolinking will can partially resolve this because it will generate all new targets, but I think the solution files will get a new project entry along with a their old dead one.

The workaround on the module developers side is to make sure they don't change the guid (and revert such changes if using `react-native-windows-init`).

One solution would be to improve autolinking to support "clearing out" old/missing autolinked projects from a solution file, but that would mean being able to identify such projects accurately.

Another solution might be to update our template generation code to ensure we don't change project guids in existing projects.

Ultimately using `react-native-windows-init` to update any project (app or module) is a workaround for not having a better upgrade experience, but it does do a lot of the work already.

### PR DESCRIPTION
**Pull request by TatianaKapos**
## Description
If there is an existing project that react-native-init is going to overwrite, reuse the old projectGUID instead of generating a new one.

### Type of Change
- New feature (non-breaking change which adds functionality)

### Why
Resolves #8621
TLDR - creating a new projectGUID can create issues when upgrading lib projects by creating unnecesary changes in the project and solution file.

### What
Checks if we're overwriting a project file, extracts the projectGUID and uses that in the template

## Testing
Tested locally if projectGUID gets overwritten or not

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9453)

### CODE REVIEW

File: packages/@react-native-windows/cli/src/generator-windows/index.ts [hunk 1]
```diff
@@ -95,6 +95,34 @@ export async function copyProjectTemplateAndReplace(
     namespace = namespace.split(/[.:]+/).map(pascalCase).join('.');
   }
 
+  // Checking if we're overwriting an existing project and re-uses their projectGUID
```
Comment (NickGerleman) [inline packages/@react-native-windows/cli/src/generator-windows/index.ts:98]: This function is already really hairy. Could we pull this logic out into something separate?

File: packages/@react-native-windows/cli/src/generator-windows/index.ts [hunk 2]
```diff
@@ -95,6 +95,34 @@ export async function copyProjectTemplateAndReplace(
...
+    newProjectName + '.vcxproj',
+  );
+  const projectPathCs = path.join(
+    windowsDir,
+    newProjectName,
+    newProjectName + '.csproj',
+  );
+  let reuseProjectGuid;
+  if (fs.existsSync(projectPathVcx)) {
+    console.log('Found exisitng C++ project, using old projectGUID');
```
Comment (NickGerleman) [inline packages/@react-native-windows/cli/src/generator-windows/index.ts:111]: nit: a log statement below ends with a period. We should be consistent between statements.

File: packages/@react-native-windows/cli/src/generator-windows/index.ts [hunk 3]
```diff
@@ -95,6 +95,34 @@ export async function copyProjectTemplateAndReplace(
...
+    windowsDir,
+    newProjectName,
+    newProjectName + '.vcxproj',
+  );
+  const projectPathCs = path.join(
+    windowsDir,
+    newProjectName,
+    newProjectName + '.csproj',
+  );
+  let reuseProjectGuid;
```
Comment (jonthysell) [inline packages/@react-native-windows/cli/src/generator-windows/index.ts:109]: I think something like previous, or existing projectGuid would make more sense here.

```suggestion
  let existingProjectGuid : string | undefined = undefined;
```

File: packages/@react-native-windows/cli/src/generator-windows/index.ts [hunk 4]
```diff
@@ -95,6 +95,34 @@ export async function copyProjectTemplateAndReplace(
     namespace = namespace.split(/[.:]+/).map(pascalCase).join('.');
   }
 
+  // Checking if we're overwriting an existing project and re-uses their projectGUID
+  const projectPathVcx = path.join(
```
Comment (jonthysell) [inline packages/@react-native-windows/cli/src/generator-windows/index.ts:99]: Don't forget the root `destPath`. Also, rather  than creating two path variables, this could be simplified to use the language option passed by the caller, to something like:

```js
const existingProjectPath = path.join(destPath, windowsDir, newProjectName, newProjectName + (language === 'cs' ? '.csproj' : '.vcxproj'));
```

File: packages/@react-native-windows/cli/src/generator-windows/index.ts [hunk 5]
```diff
@@ -111,10 +139,11 @@ export async function copyProjectTemplateAndReplace(
   if (options.useWinUI3) {
     console.log('Using experimental WinUI3 dependency.');
   }
+
   const projDir = 'proj';
   const srcPath = path.join(srcRootPath, `${language}-${projectType}`);
   const sharedPath = path.join(srcRootPath, `shared-${projectType}`);
-  const projectGuid = uuid.v4();
+  const projectGuid = reuseProjectGuid || uuid.v4();
```
Comment (jonthysell) [inline packages/@react-native-windows/cli/src/generator-windows/index.ts:146]:
```suggestion
  const projectGuid = existingProjectGuid || uuid.v4();
```

File: packages/@react-native-windows/cli/src/generator-windows/index.ts [hunk 6]
```diff
@@ -95,6 +95,34 @@ export async function copyProjectTemplateAndReplace(
...
+    newProjectName,
+    newProjectName + '.vcxproj',
+  );
+  const projectPathCs = path.join(
+    windowsDir,
+    newProjectName,
+    newProjectName + '.csproj',
+  );
+  let reuseProjectGuid;
+  if (fs.existsSync(projectPathVcx)) {
```
Comment (jonthysell) [inline packages/@react-native-windows/cli/src/generator-windows/index.ts:110]: If you only have the one `existingProjectPath` from above, you can simplify this to a single if statement. Just generalize the logging to be "Found existing project, extracting ProjectGuid."

**Review by jonthysell — CHANGES_REQUESTED** [review]

File: packages/@react-native-windows/cli/src/generator-windows/index.ts (hunk 4 above)
Comment (TatianaKapos) [reply packages/@react-native-windows/cli/src/generator-windows/index.ts]: oh good call out, I totally forgot about the language option! :)

File: packages/@react-native-windows/cli/src/generator-windows/index.ts (hunk 1 above)
Comment (TatianaKapos) [reply packages/@react-native-windows/cli/src/generator-windows/index.ts:99]: Per Jon's comments, this logic has been simplified to a single if-statment, let me know if you still think it should be pulled out into another function! :)

**Review by jonthysell — APPROVED** [review]

**Review by NickGerleman — APPROVED** [review]
